snakemake/remote/GS.py

```diff
@@ -125,6 +126,27 @@ def __init__(
         self._bucket = None
         self._blob = None
 
+    def inventory(self, cache: snakemake.io.IOCache):
+        """Using client.list_blobs(), we want to iterate over the objects in
+           the "folder" of a bucket and store information about the IOFiles in the
+           provided cache (snakemake.io.IOCache) indexed by bucket/blob name.
+           This will be called by the first mention of a remote object, and
+           iterate over the entire bucket once (and then not need to again). 
+           This includes:
+            - cache.exist_remote
+            - cache_mtime
+            - cache.size
+        """
+        for blob in self.client.list_blobs(
+            self.bucket_name, prefix=os.path.dirname(self.blob.name)
+        ):
+
+            # By way of being listed, it exists. mtime is a datetime object
+            name = "%s/%s" % (blob.bucket.name, blob.name)
```

Comment:
Please use the .format method instead, old python string formatting might go away at some point.

Reply:
All set.